pkg/encryption/path.go

```diff
@@ -159,3 +164,59 @@ func decryptPathComponent(comp string, cipher storj.CipherSuite, key *storj.Key)
 
 	return string(decrypted), nil
 }
+
+func encodeSegment(segment []byte) []byte {
+	if len(segment) == 0 {
+		return emptyComponent
+	}
+
+	result := make([]byte, 0, len(segment)*2+1)
+	result = append(result, notEmptyComponentPrefix)
+	for i := 0; i < len(segment); i++ {
+		switch {
+		case segment[i] == escape1:
+			result = append(result, []byte{escape1, 1}...)
+		case segment[i] == escape1+1:
+			result = append(result, []byte{escape1, 2}...)
+		case segment[i] == escape2:
+			result = append(result, []byte{escape2, 1}...)
+		case segment[i] == escape2+1:
+			result = append(result, []byte{escape2, 2}...)
+		case segment[i] == escape3-1:
+			result = append(result, []byte{escape3, 1}...)
+		case segment[i] == escape3:
+			result = append(result, []byte{escape3, 2}...)
+		default:
+			result = append(result, segment[i])
+		}
+	}
+	return result
+}
+
+func decodeSegment(segment []byte) []byte {
+	if len(segment) == 0 {
+		return segment
+	}
+	if segment[0] == emptyComponent[0] {
+		return []byte{}
+	}
+
+	// TODO should first byte different than x02 should be invalid?
```

Comment:
I think that sequences that aren't properly escaped should be errors when attempting to unescape.

Reply:
done